qiskit/circuit/random/utils.py

```diff
@@ -11,15 +11,415 @@
+        (standard_gates.CU3Gate, 2, 3),
+        (standard_gates.CYGate, 2, 0),
+        (standard_gates.CZGate, 2, 0),
+        (standard_gates.RXXGate, 2, 1),
+        (standard_gates.RYYGate, 2, 1),
+        (standard_gates.RZZGate, 2, 1),
+        (standard_gates.RZXGate, 2, 1),
+        (standard_gates.XXMinusYYGate, 2, 2),
+        (standard_gates.XXPlusYYGate, 2, 2),
+        (standard_gates.ECRGate, 2, 0),
+        (standard_gates.CSGate, 2, 0),
+        (standard_gates.CSdgGate, 2, 0),
+        (standard_gates.SwapGate, 2, 0),
+        (standard_gates.iSwapGate, 2, 0),
+    ]
+    gates_3q = [
+        (standard_gates.CCXGate, 3, 0),
+        (standard_gates.CSwapGate, 3, 0),
+        (standard_gates.CCZGate, 3, 0),
+        (standard_gates.RCCXGate, 3, 0),
+    ]
+    gates_4q = [
+        (standard_gates.C3SXGate, 4, 0),
+        (standard_gates.RC3XGate, 4, 0),
+    ]
+    return (gates_1q, gates_2q, gates_3q, gates_4q)
+
+
+def random_circuit_from_graph(
+    interaction_graph,
+    min_2q_gate_per_edge,
+    max_operands=2,
+    measure=False,
+    conditional=False,
+    reset=False,
+    seed=None,
+    insert_1q_oper=False,
+    prob_conditional: float = 0.1,
+):
+    """Generate random circuit of arbitrary size and form which strictly respect the interaction
+    graph passed as argument. Interaction Graph is a graph G=(V, E) where V are the qubits in the
+    circuit, and, E is the set of two-qubit gate interactions between two particular qubits in the
+    circuit.
+
+    This function will generate a random circuit by randomly selecting gates from the set of
+    standard gates in :mod:`qiskit.circuit.library.standard_gates`. User can attach a numerical
+    value as a metadata to the edge of the graph indicating the edge weight for that particular edge,
+    These edge weights  would be normalized to the probabilities for the edges of getting selected.
+    If all the edge weights are passed as `None`, then the probability of each qubit-pair of getting
+    selected is set to 1/N. (where N is the number of edges in the interaction_graph passed in)
+
+    If numerical values are present as probabilities but some/any are None, or zero, or negative,
+    this will raise a ValueError.
+
+    If :arg:`max_operands` is set to 1, then there are no 2Q operations, so no need to take care
+    of the edges, in such case the function will return a circuit from the `random_circuit` function,
+    which would be passed with the :arg:`max_operands` as 1.
+
+    If :arg:`max_operands` is set to 2, then in every iteration `N` 2Q gates and qubit-pairs
+    are chosen at random, the qubit-pairs are also chosen at random based on the probability
+    attached to the qubit-pair, the 2Q gates are applied on the qubit-paris on which are no gates
+    already present for that particular iteration, this is to make sure that for a particular
+    iteration only one circuit layer exists, now for those qubits for this particular iteration on
```

Comment:
1>> extract qubit-pairs from input interaction graph

2>> Iterate over this qubit-pair, apply 2Q gates but make sure those 2Q gates are
only applied to those qubit-pair which are idle(both control and target), in case
any control, target found not to be idle, skip to another qubit-pair.

3>> Now, if any qubit still remains idle, apply 1Q if `insert_1Q_oper` is enabled.

A qubit which has only reset applied on it is considered idle.
Anyways, provided a better explanation in the docstring.